kernel/frame_allocator/Cargo.toml

```diff
@@ -8,10 +8,12 @@ edition = "2021"
 [dependencies]
 log = "0.4.8"
 spin = "0.9.4"
-intrusive-collections = "0.9.0"
 static_assertions = "1.1.0"
 
 range_inclusive = { path = "../../libs/range_inclusive" }
 
 kernel_config = { path = "../kernel_config" }
 memory_structs = { path = "../memory_structs" }
+
+[dependencies.intrusive-collections]
+git = "https://github.com/Ramla-I/intrusive-rs"
```

Comment:
added `get_mut`, and other minimal changes required to support that function.

Reply:
I don't know if implementing `get_mut()` on an RBTree's `CursorMut` is sound. We'd need to submit that as a separate PR to the upstream `intrusive-rs` crate to get feedback from the maintainers to see if doing that is correct. I have a sneaking suspicion that it's not sound to do so, otherwise Amanieu likely would've provided that API.

Reply:
ok, so there have been multiple issues filed about this ([link](https://github.com/Amanieu/intrusive-rs/issues?q=is%3Aissue+get_mut+is%3Aclosed))
It's considered unsound behavior because there can be multiple references to an item, and the recommendation is to use Cell/RefCell

Reply:
Ah yep, there it is. I knew I remembered seeing discussions about it in the past.

I think replace_with() is good enough for our needs here.